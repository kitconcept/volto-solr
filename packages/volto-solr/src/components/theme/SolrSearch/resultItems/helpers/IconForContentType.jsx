import { Icon } from '@plone/volto/components';
import config from '@plone/volto/registry';

const IconForContentType = ({ type }) => (
  <Icon
    className="itemIcon"
    size="20px"
    name={
      config.settings.contentTypeSearchResultIcons[type] ||
      config.settings.contentTypeSearchResultDefaultIcon
    }
  />
);

export default IconForContentType;
