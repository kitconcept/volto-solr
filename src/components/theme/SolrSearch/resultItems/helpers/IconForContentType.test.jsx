import { create } from 'react-test-renderer';
import IconForContentType from './IconForContentType';
import config from '@plone/volto/registry';

describe('IconForContentType', () => {
  let origContentTypeSearchResultIcons;
  let origContentTypeSearchResultDefaultIcon;

  beforeEach(() => {
    origContentTypeSearchResultIcons =
      config.settings.contentTypeSearchResultIcons;
    origContentTypeSearchResultDefaultIcon =
      config.settings.contentTypeSearchResultDefaultIcon;
    config.settings.contentTypeSearchResultIcons = {
      Event: { content: `calendarSVG` },
      Image: { content: `imageSVG` },
    };
    config.settings.contentTypeSearchResultDefaultIcon = { content: 'fileSVG' };
  });

  afterEach(() => {
    config.settings.contentTypeSearchResultIcons = origContentTypeSearchResultIcons;
    config.settings.contentTypeSearchResultDefaultIcon = origContentTypeSearchResultDefaultIcon;
  });

  test('default icon', () => {
    const component = create(<IconForContentType type={'Foo'} />);
    const rendered = component.toJSON();
    expect(rendered).toMatchSnapshot();
  });

  test('Event icon', () => {
    const component = create(<IconForContentType type={'Event'} />);
    const rendered = component.toJSON();
    expect(rendered).toMatchSnapshot();
  });

  test('Image icon', () => {
    const component = create(<IconForContentType type={'Image'} />);
    const rendered = component.toJSON();
    expect(rendered).toMatchSnapshot();
  });
});
