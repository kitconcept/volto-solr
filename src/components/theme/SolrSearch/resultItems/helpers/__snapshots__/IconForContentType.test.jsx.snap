// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`IconForContentType Event icon 1`] = `
<svg
  className="icon itemIcon"
  dangerouslySetInnerHTML={
    Object {
      "__html": "calendarSVG",
    }
  }
  onClick={null}
  style={
    Object {
      "fill": "currentColor",
      "height": "20px",
      "width": "auto",
    }
  }
/>
`;

exports[`IconForContentType Image icon 1`] = `
<svg
  className="icon itemIcon"
  dangerouslySetInnerHTML={
    Object {
      "__html": "imageSVG",
    }
  }
  onClick={null}
  style={
    Object {
      "fill": "currentColor",
      "height": "20px",
      "width": "auto",
    }
  }
/>
`;

exports[`IconForContentType default icon 1`] = `
<svg
  className="icon itemIcon"
  dangerouslySetInnerHTML={
    Object {
      "__html": "fileSVG",
    }
  }
  onClick={null}
  style={
    Object {
      "fill": "currentColor",
      "height": "20px",
      "width": "auto",
    }
  }
/>
`;
